instance/instance.go

```diff
@@ -140,20 +165,17 @@ func (h *Instance) ExactImages(images []flux.ImageID) (ImageMap, error) {
 
 // Checks whether the given image exists in the repository.
 // Return true if exist, false otherwise
-func (h *Instance) imageExists(image flux.ImageID) (exists bool, err error) {
-	// Get a list of images
-	images, err := h.registry.GetRepository(image.Repository())
+func (h *Instance) imageExists(image flux.ImageID) (bool, error) {
+	img, err := registry.ParseImage(string(image), nil)
```

Comment:
Yes, I tried that. After about a day of attempting, I gave up. That ImageId string is in nearly every file in the repo. I.e. nearly every file is directly coupled to flux package through this.

And there are major issues with changing it. Since some files expect it not to have a tag, other expect it to have a tag. I think I've even seens places that expect it to be with/without the host.

So, I could go through and change everything all at once, but it would be a massive PR and would probably break everything.